lib/src/components/NxToast/NxToastContainer.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useRef } from 'react';
+import { createPortal } from 'react-dom';
+import NxToast from './NxToast';
+import { NxToastContainerProps, nxToastContainerPropTypes } from './types';
+import classNames from 'classnames';
+
+const NxToastContainer = (props: NxToastContainerProps) => {
+  const { toasts } = props,
+      className = classNames('nx-toast__wrapper'),
+      toastContainerRef = useRef<HTMLDivElement | null>(null);
+
+  const [domReady, setDomReady] = React.useState(false);
+
+  useEffect(() => {
+    setDomReady(true);
+  }, []);
+
+  //Create a portal to .nx-page-main if exists, if not, default to document.body
+  const domNodeForPortal = document.querySelector('.nx-page-main') as HTMLElement || document.body;
+
+  if (!domReady) {
+    return null;
```

Comment:
What is the purpose of this mechanism?

Reply:
I believe `!domready` is there in case `NxToastContainer` loads up before other parts of the DOM do, such as `NxToastProvider`,`.nx-page-main`, etc...

Reply:
> What is the purpose of this mechanism?

>I believe !domready helps NxToastContainer avoid bugs in case it loads up before other parts of the DOM do, such as NxToastProvider,.nx-page-main, etc...

Portal and !domReady removed here: [9540220](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/9540220b0a33eafa70152d5de05ed7c8763a6cb2)